pandas/_libs/lib.pyx

```diff
@@ -1319,8 +1320,11 @@ def infer_dtype(value: object, skipna: object=None) -> str:
             return 'boolean'
 
     elif isinstance(val, str):
-        if is_string_array(values, skipna=skipna):
-            return 'string'
+        if is_string_array(values, skipna=True):
```

Comment:
It's unclear to me whether `mixed-string` is actually a useful distinction from just `string`. As an alternative, I can just always return `"string"` here, but that might break code relying on a mix of strings and NAs being "mixed".

Reply:
I'm trying out just "string" now. Will see what break.s

Reply:
The same is true for `mixed-integer` no? (so we should be consistent?) But isn't that the purpose of passing `skipna=True` to not see missing values as non-strings?